shared/domains/chelonia/chelonia.js

```diff
@@ -645,11 +654,12 @@ export default (sbp('sbp/selectors/register', {
         [`${contract.manifest}/${action}/process`]: (message: Object, state: Object) => {
           const { meta, data, contractID } = message
           // TODO: optimize so that you're creating a proxy object only when needed
-          const gProxy = gettersProxy(state, contract.getters)
-          state = state || contract.state(contractID)
-          contract.metadata.validate(meta, { state, ...gProxy, contractID })
-          contract.actions[action].validate(data, { state, ...gProxy, meta, message, contractID })
-          contract.actions[action].process(message, { state, ...gProxy })
+          const stateCopy = cloneDeep(state || contract.state(contractID))
```

Comment:
Why is this change being made? Ultimately `state` gets modified on line 662 anyway... so why this clone/copy?

Reply:
This is done to emulate the sandbox environment, where contracts don't have direct access to modify the state. I made this change to test that things didn't break or to fix it as appropriate (remember, when I started working on this it was the SW + sandboxing). It's probably fine to remove this, other than that contracts need to be re-tested when sandboxing is implemented. On a similar note, the same thing applies to the proxy (I didn't change it, but it might need to be done differently).

Reply:
TL;DR: Can be reverted.

Reply:
OK, yeah I would revert this and save it for when we get to the sandbox

Reply:
Done